src/bucket/Bucket.cpp

```diff
@@ -78,15 +78,27 @@ Bucket::Bucket()
 }
 
 XDRInputFileStream&
-Bucket::getStream()
+Bucket::getIndexStream()
 {
-    if (!mStream)
+    if (!mIndexStream)
     {
-        mStream = std::make_unique<XDRInputFileStream>();
+        mIndexStream = std::make_unique<XDRInputFileStream>();
         releaseAssertOrThrow(!mFilename.empty());
-        mStream->open(mFilename.string());
+        mIndexStream->open(mFilename.string());
     }
-    return *mStream;
+    return *mIndexStream;
+}
+
+XDRInputFileStream&
+Bucket::getEvictionStream()
+{
+    if (!mEvictionStream)
+    {
+        mEvictionStream = std::make_unique<XDRInputFileStream>();
+        releaseAssertOrThrow(!mFilename.empty());
+        mEvictionStream->open(mFilename.string());
```

Comment:
IIUC the intention is to open two streams on the same file, right? A helper would be useful here then to avoid duplication and improve clarity.